[stable/metrics-server] Allow for tls configuration options

dioguerra opened the pull request:
This patch adds a tlcConfig field where we can specify:
* caFile path
* certFile path
* keyFile path
* insecureSkipVerify

Enabling the metrics-server pod to communicate Non-Anonymously with the kubernetes resources

Signed-off-by: Diogo Guerra <dy090.guerra@gmail.com>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
This patch adds a tlcConfig field where we can specify:
* caFile path
* certFile path
* keyFile path
* insecureSkipVerify

Enabling the metrics-server pod to communicate Non-Anonymously with the kubernetes resources

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

Enables the metrics-server pod to comunicate securely with the cluster

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Variables are documented in the README.md

dioguerra committed:
[stable/metrics-server] Allow for tls configuration

This patch adds a tlcConfig field where we can specify:
* caFile path
* certFile path
* keyFile path
* insecureSkipVerify

Enabling the metrics-server pod to communicate Non-Anonymously with the kubernetes resources

Signed-off-by: Diogo Guerra <dy090.guerra@gmail.com>

[FIX] default values to proper settings

Signed-off-by: Diogo Guerra <dy090.guerra@gmail.com>

olemarkus commented on stable/metrics-server/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 description: Metrics Server is a cluster-wide aggregator of resource usage data.
 name: metrics-server
-version: 2.1.0
+version: 2.1.1
```
This needs a minor version bump

dioguerra replied:
In the appVersion?

olemarkus replied:
No, the chart version. Introducting new features requires minor version bump. patch version is only for bugfixes and similar

olemarkus commented on stable/metrics-server/README.md:
```diff
@@ -19,3 +19,7 @@ Parameter | Description | Default
 `nodeSelector` | Node labels for pod assignment | `{}`
 `affinity` | Node affinity | `{}`
 `replicas` | Number of replicas | `1`
+`tlsConfig.caFile` | caFile directory path | ``
```
Can you rename tlsConfig to just tls? Config postfix feels a bit redundant.

olemarkus commented on stable/metrics-server/templates/metrics-server-deployment.yaml:
```diff
@@ -26,13 +26,65 @@ spec:
+            - --tls-cert-file
+            - /certs/certFile
+            {{- end}}
+            {{- if .Values.tlsConfig.keyFile }}
```
keyFile should be rename to privateKeyFile to match the flag

olemarkus commented on stable/metrics-server/templates/metrics-server-deployment.yaml:
```diff
@@ -26,13 +26,65 @@ spec:
+            - --tls-private-key-file
+            - /certs/keyFile
+            {{- end}}
+            {{- if .Values.tlsConfig.insecureSkipVerify }}
```
This one shouldn't be a part of this config as it can be set independently of the other TLS settings

dioguerra replied:
insecureSkipVerify can be outside the  tls group but should i group them in a connection group?

I followed the prometheus-operator etcd example, maybe not the best one. I'm now making the changes to be similar to the prometheus-adapter as advised. will bring the insecureSkipVerify to outside and well see where we go from there

dioguerra replied:
it feels unnatural to be outside of the tls group as the flag it posts specifically says> - --kubelet-insecure-tls

olemarkus commented on stable/metrics-server/values.yaml:
```diff
@@ -36,3 +36,9 @@ tolerations: []
+#   caFile:   /etc/kubernetes/certs/ca.crt
+#   certFile: /etc/kubernetes/certs/kubelet.crt
+#   keyFile:  /etc/kubernetes/certs/kubelet.key
+#   insecureSkipVerify: true
```
Make sure you take the comment and config on line 27-28 into consideration too, as this would replace that warning.

dioguerra committed:
[stable/metrics-server] Allow for tls configuration

This patch adds a tlcConfig field where we can specify:
* caFile path
* certFile path
* keyFile path
* insecureSkipVerify

Enabling the metrics-server pod to communicate Non-Anonymously with the kubernetes resources

Signed-off-by: Diogo Guerra <dy090.guerra@gmail.com>

[FIX] default values to proper settings

Signed-off-by: Diogo Guerra <dy090.guerra@gmail.com>
